src/main/webapp/WEB-INF/web.xml

```diff
@@ -14,5 +16,21 @@
     <servlet-name>productList</servlet-name>
     <url-pattern>/products</url-pattern>
   </servlet-mapping>
+  
+  <listener>
+      <listener-class>
+          com.es.phoneshop.listener.ProductContextListener
+      </listener-class>
+  </listener>
+  
+  <servlet>
+    <servlet-name>productDetails</servlet-name>
+    <servlet-class>com.es.phoneshop.web.ProductDetailsServlet</servlet-class>
+  </servlet>
+  
```

Comment:
<error-page> here?

Task "Display 404 page with “Product with code xxx not found” message if a product can’t be found."

Reply:
oooo, I forgot about it